src/main/scala/scaladl/layers/Layer.scala

```diff
@@ -384,6 +423,26 @@ object FeedForwardTopology {
     }
     FeedForwardTopology(layers)
   }
+
+  /**
+   * Creates a multi-layer perceptron for regression
+   *
+   * @param layerSizes sizes of layers including input and output size
+   * @return multilayer perceptron topology
+   */
+  def multiLayerPerceptronRegression(layerSizes: Array[Int]): FeedForwardTopology = {
+    val layers = new Array[Layer]((layerSizes.length - 1) * 2)
+    for (i <- 0 until layerSizes.length - 1) {
+      layers(i * 2) = new AffineLayer(layerSizes(i), layerSizes(i + 1))
+      layers(i * 2 + 1) =
+        if (i == layerSizes.length - 2) {
+          new LinearLayerWithSquaredError()
```

Comment:
Hi @avulanov and @JeremyNixon in order to refactor code I think that is a good idea add a typelayer to function:

    val layers = Array[Int](2, 5, 2)

    val layerstype = Array[String]("sigmoid", "softmaxOnTop")

    val trainer = new MultilayerPerceptronClassifier()

If you can try, you can clone this repo: https://github.com/mhaut/scalable-deeplearning